File: DevExpress-Demo/VB/GridDemo.Wpf/ModuleResources/CollectionViewViewModel.vb
Imports DevExpress.Mvvm
Imports DevExpress.Xpf.DemoBase.DataClasses
Imports System.Collections
Imports System.ComponentModel
Imports System.Windows.Data

Namespace GridDemo

    Public Class CollectionViewViewModel
        Inherits BindableBase

        Private _CollectionView As ICollectionView

        Private employeesCore As IList = EmployeesWithPhotoData.DataSource

        Public ReadOnly Property Employees As IList
            Get
                Return employeesCore
            End Get
        End Property

        Public Property CollectionView As ICollectionView
            Get
                Return _CollectionView
            End Get

            Private Set(ByVal value As ICollectionView)
                _CollectionView = value
            End Set
        End Property

        Public Sub New()
            CollectionView = New CollectionViewSource() With {.Source = Employees}.View
            CollectionView.GroupDescriptions.Add(New PropertyGroupDescription("JobTitle"))
            CollectionView.SortDescriptions.Add(New SortDescription("JobTitle", ListSortDirection.Ascending))
            CollectionView.MoveCurrentToFirst()
        End Sub
    End Class
End Namespace

File: DevExpress-Demo/CS/GridDemo.Wpf/ModuleResources/CollectionViewViewModel.cs
using DevExpress.Mvvm;
using DevExpress.Xpf.DemoBase.DataClasses;
using System.Collections;
using System.ComponentModel;
using System.Windows.Data;

namespace GridDemo {
    public class CollectionViewViewModel : BindableBase {
        IList employeesCore = EmployeesWithPhotoData.DataSource;
        public IList Employees { get { return employeesCore; } }
        public ICollectionView CollectionView { get; private set; }

        public CollectionViewViewModel() {
            CollectionView = new CollectionViewSource() { Source = Employees }.View;
            CollectionView.GroupDescriptions.Add(new PropertyGroupDescription("JobTitle"));
            CollectionView.SortDescriptions.Add(new SortDescription("JobTitle", ListSortDirection.Ascending));
            CollectionView.MoveCurrentToFirst();
        }
    }
}
